node/runtime/src/system.rs

```diff
@@ -189,6 +188,45 @@ pub fn swap_key(
     Ok(vec![])
 }
 
+pub fn add_key(
+    state_update: &mut StateDbUpdate,
+    body: &AddKeyTransaction,
+    account: &mut Account
+) -> Result<Vec<ReceiptTransaction>, String> {
+    let new_key = PublicKey::new(&body.new_key)?;
+    account.public_keys.push(new_key);
```

Comment:
We need to check the number of public_keys and prevent adding more than the limit.
Also we need to check that the public_key doesn't exist yet

Reply:
Fixed